nrt/nrt.py

```diff
@@ -612,14 +613,41 @@ def main():
         end_time = time.time()
         NRT.setRealTime(end_time - start_time)
 
+        maximumNumberOfThreads = int(os.getenv ('DYNAWO_NB_PROCESSORS_USED'))
+        threads_list = []
+        dir_list = []
+        semaphore = threading.Semaphore (maximumNumberOfThreads)
+        pool = nrtDiff.ActivePool()
         for case in NRT.test_cases_:
             if (not case.ok_):
                 case.diff_ = nrtDiff.UNABLE_TO_CHECK
             else:
                 case_dir = os.path.dirname (case.jobs_file_)
-                (status, messages) = nrtDiff.DirectoryDiffReferenceDataJob (case_dir)
-                case.diff_ = status
-                case.diff_messages_ = messages
+                if case_dir in dir_list: continue
+                dir_list.append(case_dir)
+                thread = threading.Thread (target = nrtDiff.DirectoryDiffReferenceDataJobMultiThread, name = case_dir, args = (case, semaphore, pool))
+                thread.setDaemon(True)
+                threads_list.append(thread)
+                thread.start()
+        #Keep the main thread alive while threads are running and catch interruptions
+        try:
```

Comment:
Isn't it possible to wait for all thread to end instead of polling status (like promises) ?

Reply:
this was the initial implementation but it was disabling the possibility to use Ctrl-C during the comparison. This implementation with Daemon allows to have both